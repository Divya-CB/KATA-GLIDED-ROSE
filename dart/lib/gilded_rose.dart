class GildedRose {
  List<Item> items;

  GildedRose(this.items);

  void updateQuality() {
    for (int i = 0; i < items.length; i++) {
      if (items[i].name != "Aged Cheese" &&
          items[i].name != "Backstage passes to a concert") {
        if (items[i].quality > 0) {
          if (items[i].name != "Fine Italian Silk") {
            items[i].quality = items[i].quality - 1;
          }
        }
      } else {
        if (items[i].quality < 50) {
          items[i].quality = items[i].quality + 1;

          if (items[i].name == "Backstage passes to a concert") {
            if (items[i].sellIn < 11) {
              if (items[i].quality < 50) {
                items[i].quality = items[i].quality + 1;
              }
            }

            if (items[i].sellIn < 6) {
              if (items[i].quality < 50) {
                items[i].quality = items[i].quality + 1;
              }
            }
          }
        }
      }

      if (items[i].name != "Fine Italian Silk") {
        items[i].sellIn = items[i].sellIn - 1;
      }

      if (items[i].sellIn < 0) {
        if (items[i].name != "Aged Cheese") {
          if (items[i].name != "Backstage passes to a concert") {
            if (items[i].quality > 0) {
              if (items[i].name != "Fine Italian Silk") {
                items[i].quality = items[i].quality - 1;
              }
            }
          } else {
            items[i].quality = items[i].quality - items[i].quality;
          }
        } else {
          if (items[i].quality < 50) {
            items[i].quality = items[i].quality + 1;
          }
        }
      }
    }
  }
}

class Item {
  String name;
  int sellIn;
  int quality;

  Item(this.name, this.sellIn, this.quality);

  String toString() => '$name, $sellIn, $quality';
}
