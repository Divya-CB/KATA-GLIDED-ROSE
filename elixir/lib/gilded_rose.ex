defmodule GildedRose do

  def update_quality(items) do
    Enum.map(items, &update_item_new/1)
  end

  @ticket "Backstage passes to a TAFKAL80ETC concert"
  def update_item_new(item = %{ name: @ticket , sell_in: days, quality: quality }) when days - 1 < 0, do:
    %{item | sell_in: days - 1, quality: 0 }

  def update_item_new(item = %{ name: @ticket , sell_in: days, quality: quality }) when days - 1 <= 5, do:
    %{item | sell_in: days - 1, quality: updated_quality(50, quality + 3) }

  def update_item_new(item = %{ name: @ticket , sell_in: days, quality: quality }) when days - 1 <= 10, do:
    %{item | sell_in: days - 1, quality: updated_quality(50, quality + 2) }

  def update_item_new(item = %{ name: @ticket , sell_in: days, quality: quality }), do:
    %{item | sell_in: days - 1, quality: updated_quality(50, quality + 1) }


  # def update_item_new(item = %{ name: @ticket , sell_in: days, quality: quality }) do
  #   left = days - 1 

  #   cond do
  #     left > 5 && left < 10 ->  %{item | sell_in: left, quality: updated_quality(50, quality + 2) }
  #     left > 0 && left < 5 ->   %{item | sell_in: left, quality: updated_quality(50, quality + 3) }
  #     left < 0 ->               %{item | sell_in: left, quality: 0 }
  #     true ->                   %{item | sell_in: left, quality: updated_quality(50, quality + 1) }
  #   end
  # end

  @brie "Aged Brie"
  def update_item_new(item = %{ name: @brie, sell_in: days, quality: quality }) when  days - 1 < 0,  do:
    %{item | sell_in: days - 1, quality: updated_quality(50, quality + 2) }

  def update_item_new(item = %{ name: @brie, sell_in: days, quality: quality }), do:
    %{item | sell_in: days - 1, quality: updated_quality(50, quality + 1) }
    

  @sulfuras "Sulfuras, Hand of Ragnaros"
  def update_item_new(item = %{ name: @sulfuras, quality: quality }) when quality > 80, do: %{item | quality: 80 }
  def update_item_new(item = %{ name: @sulfuras }), do: item


  defp updated_quality(max, new_amount) when new_amount > max, do: max
  defp updated_quality(_, new_amount), do: new_amount






  def update_item(item) do
    item = cond do
      item.name != "Aged Brie" && item.name != "Backstage passes to a TAFKAL80ETC concert" ->
        if item.quality > 0 do
          if item.name != "Sulfuras, Hand of Ragnaros" do
            %{item | quality: item.quality - 1}
          else
            item
          end
        else
          item
        end
      true ->
        cond do
          item.quality < 50 ->
            item = %{item | quality: item.quality + 1}
            cond do
              item.name == "Backstage passes to a TAFKAL80ETC concert" ->
                item = cond do
                  item.sell_in < 11 ->
                    cond do
                      item.quality < 50 ->
                        %{item | quality: item.quality + 1}
                      true -> item
                    end
                  true -> item
                end
                cond do
                  item.sell_in < 6 ->
                    cond do
                      item.quality < 50 ->
                        %{item | quality: item.quality + 1}
                      true -> item
                    end
                  true -> item
                end
              true -> item
            end
          true -> item
        end
    end
    item = cond do
      item.name != "Sulfuras, Hand of Ragnaros" ->
        %{item | sell_in: item.sell_in - 1}
      true -> item
    end
    cond do
      item.sell_in < 0 ->
        cond do
          item.name != "Aged Brie" ->
            cond do
              item.name != "Backstage passes to a TAFKAL80ETC concert" ->
                cond do
                  item.quality > 0 ->
                    cond do
                      item.name != "Sulfuras, Hand of Ragnaros" ->
                        %{item | quality: item.quality - 1}
                      true -> item
                    end
                  true -> item
                end
              true -> %{item | quality: item.quality - item.quality}
            end
          true ->
            cond do
              item.quality < 50 ->
                %{item | quality: item.quality + 1}
              true -> item
            end
        end
      true -> item
    end
  end
end
