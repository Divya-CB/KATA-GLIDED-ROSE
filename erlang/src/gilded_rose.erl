-module(gilded_rose).

-include("gilded_rose.hrl").

-export([
  update_quality/1
]).

-spec update_quality([#item{}]) -> [#item{}].
update_quality(Items) ->
  lists:map(fun update_item/1, Items).

-spec update_item(#item{}) -> #item{}.
update_item(Item = #item{name = Name}) ->
  Item1 = if
            Name /= "Aged Cheese" andalso Name /= "Backstage passes to a concert" ->
              if
                Item#item.quality > 0 ->
                  if
                    Name /= "Fine Italian Silk" ->
                      Item#item{quality = Item#item.quality - 1};
                    true ->
                      Item
                  end;
                true ->
                  Item
              end;
            true ->
              if
                Item#item.quality < 50 ->
                  Item2 = Item#item{quality = Item#item.quality + 1},
                  if
                    Name == "Backstage passes to a concert" ->
                      Item3 = if
                                Item2#item.sell_in < 11 ->
                                  if
                                    Item2#item.quality < 50 ->
                                      Item2#item{quality = Item2#item.quality + 1};
                                    true -> Item2
                                  end;
                                true -> Item2
                              end,
                      if
                        Item3#item.sell_in < 6 ->
                          if
                            Item3#item.quality < 50 ->
                              Item3#item{quality = Item3#item.quality + 1};
                            true -> Item3
                          end;
                        true -> Item3
                      end;
                    true -> Item2
                  end;
                true -> Item
              end
          end,
  Item4 = if
            Name /= "Fine Italian Silk" ->
              Item1#item{sell_in = Item1#item.sell_in - 1};
            true -> Item1
          end,
  if
    Item4#item.sell_in < 0 ->
      if
        Name /= "Aged Cheese" ->
          if
            Name /= "Backstage passes to a concert" ->
              if
                Item4#item.quality > 0 ->
                  if
                    Name /= "Fine Italian Silk" ->
                      Item4#item{quality = Item4#item.quality - 1};
                    true -> Item4
                  end;
                true -> Item4
              end;
            true -> Item4#item{quality = Item4#item.quality - Item4#item.quality}
          end;
        true ->
          if
            Item4#item.quality < 50 ->
              Item4#item{quality = Item4#item.quality + 1};
            true -> Item4
          end
      end;
    true -> Item4
  end.
