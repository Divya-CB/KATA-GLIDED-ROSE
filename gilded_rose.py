class Item:
    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def __repr__(self):
        return "%s, %s, %s" % (self.name, self.sell_in, self.quality)


def is_aged_brie(item: Item) -> None:
    return item.name == "Aged Brie"


def update_item(item: Item):
    is_sulfuras = item.name == "Sulfuras, Hand of Ragnaros"
    is_backstage_pass = item.name == "Backstage passes to a TAFKAL80ETC concert"

    if is_sulfuras:
        return

    if is_aged_brie(item):
        if item.quality < 50:
            item.quality = item.quality + 1
        item.sell_in = item.sell_in - 1
        if item.sell_in < 0:
            if item.quality < 50:
                item.quality = item.quality + 1
        return

    if is_backstage_pass:
        if item.quality < 50:
            item.quality = item.quality + 1
            if item.sell_in < 11:
                if item.quality < 50:
                    item.quality = item.quality + 1
            if item.sell_in < 6:
                if item.quality < 50:
                    item.quality = item.quality + 1
        item.sell_in = item.sell_in - 1
        if item.sell_in < 0:
            item.quality = item.quality - item.quality
        return

    if item.quality > 0:
        item.quality = item.quality - 1
    item.sell_in = item.sell_in - 1
    if item.sell_in < 0:
        if item.quality > 0:
            item.quality = item.quality - 1


class GildedRose:
    def __init__(self, items: list[Item]):
        self.items = items

    def update_quality(self):
        for item in self.items:
            update_item(item)
