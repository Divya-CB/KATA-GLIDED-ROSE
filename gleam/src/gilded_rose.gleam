import cli
import gilded_rose_item.{type GildedRose, type Item, Item}
import gleam/list

pub fn update_quality(inventory: GildedRose) -> GildedRose {
  let update_quality_item = fn(item: Item) {
    let new_quality = case
      item.name != "Aged Brie"
      && item.name != "Backstage passes to a TAFKAL80ETC concert"
    {
      True -> {
        case item.quality > 0 {
          True ->
            case item.name != "Sulfuras, Hand of Ragnaros" {
              True -> item.quality - 1
              False -> item.quality
            }
          False -> item.quality
        }
      }
      False -> {
        case item.quality < 50 {
          True ->
            item.quality
            + 1
            + case item.name == "Backstage passes to a TAFKAL80ETC concert" {
              True ->
                case item.sell_in < 11 {
                  True -> {
                    case item.quality < 49 {
                      True ->
                        1
                        + case item.sell_in < 6 {
                          True ->
                            case item.quality < 48 {
                              True -> 1
                              False -> 0
                            }
                          False -> 0
                        }
                      False -> 0
                    }
                  }
                  False -> 0
                }
              False -> 0
            }
          False -> item.quality
        }
      }
    }

    let new_sell_in = case item.name != "Sulfuras, Hand of Ragnaros" {
      True -> item.sell_in - 1
      False -> item.sell_in
    }

    case new_sell_in < 0 {
      True ->
        case item.name != "Aged Brie" {
          True ->
            case item.name != "Backstage passes to a TAFKAL80ETC concert" {
              True ->
                case new_quality > 0 {
                  True ->
                    case item.name != "Sulfuras, Hand of Ragnaros" {
                      True ->
                        Item(
                          ..item,
                          sell_in: new_sell_in,
                          quality: new_quality - 1,
                        )
                      False -> Item(..item, sell_in: new_sell_in, quality: 80)
                    }
                  False ->
                    Item(..item, sell_in: new_sell_in, quality: new_quality)
                }
              False ->
                Item(
                  ..item,
                  sell_in: new_sell_in,
                  quality: new_quality - new_quality,
                )
            }
          False ->
            case new_quality < 50 {
              True ->
                Item(..item, sell_in: new_sell_in, quality: new_quality + 1)
              False -> Item(..item, sell_in: new_sell_in, quality: 50)
            }
        }
      False ->
        case item.name == "Sulfuras, Hand of Ragnaros" {
          True -> Item(..item, sell_in: new_sell_in, quality: 80)
          False ->
            case new_quality > 50 {
              True -> Item(..item, sell_in: new_sell_in, quality: 50)
              False -> Item(..item, sell_in: new_sell_in, quality: new_quality)
            }
        }
    }
  }
  list.map(inventory, update_quality_item)
}

pub fn main() {
  cli.run_cli_app(update_quality)
}
