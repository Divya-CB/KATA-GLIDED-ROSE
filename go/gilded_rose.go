package main

import "fmt"

type GildedRose struct {
	Items []Item
}

type Item struct {
	Name            string
	SellIn, Quality int
}

func main() {
	rose := GildedRose{
		[]Item{
			Item{"+5 Dexterity Vest", 10, 20},
			Item{"Aged Brie", 2, 0},
			Item{"Elixir of the Mongoose", 5, 7},
			Item{"Sulfuras, Hand of Ragnaros", 0, 80},
			Item{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
			Item{"Conjured Mana Cake", 3, 6},
		},
	}
	rose.UpdateQuality()
	fmt.Print(rose.String())
}

func (gr *GildedRose) UpdateQuality() {

	for i := 0; i < len(gr.Items); i++ {
		item := &gr.Items[i]

		if item.Name != "Aged Brie" && item.Name != "Backstage passes to a TAFKAL80ETC concert" {
			if item.Quality > 0 {
				if item.Name != "Sulfuras, Hand of Ragnaros" {
					item.Quality = item.Quality - 1
				}
			}
		} else {
			if item.Quality < 50 {
				item.Quality = item.Quality + 1
				if item.Name == "Backstage passes to a TAFKAL80ETC concert" {
					if item.SellIn < 11 {
						if item.Quality < 50 {
							item.Quality = item.Quality + 1
						}
					}
					if item.SellIn < 6 {
						if item.Quality < 50 {
							item.Quality = item.Quality + 1
						}
					}
				}
			}
		}

		if item.Name != "Sulfuras, Hand of Ragnaros" {
			item.SellIn = item.SellIn - 1
		}

		if item.SellIn < 0 {
			if item.Name != "Aged Brie" {
				if item.Name != "Backstage passes to a TAFKAL80ETC concert" {
					if item.Quality > 0 {
						if item.Name != "Sulfuras, Hand of Ragnaros" {
							item.Quality = item.Quality - 1
						}
					}
				} else {
					item.Quality = item.Quality - item.Quality
				}
			} else {
				if item.Quality < 50 {
					item.Quality = item.Quality + 1
				}
			}
		}
	}

}

func (gr *GildedRose) String() string {
	var str string
	str += "=== The Gilded Rose ===\n"
	for i, item := range gr.Items {
		str += fmt.Sprintf("%d) %s (quality %d, sell in %d)\n", i+1, item.Name, item.Quality, item.SellIn)
	}
	return str
}
