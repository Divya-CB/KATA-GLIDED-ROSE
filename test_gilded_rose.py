from gilded_rose import Item, GildedRose
from approvaltests.combination_approvals import verify_all_combinations


def test_update_quality():
    input_names = [
        "Normal",
        "Conjured",
        "Aged Brie",
        "Backstage passes to a TAFKAL80ETC concert",
        "Sulfuras, Hand of Ragnaros",
    ]
    input_qualities = [-1, 0, 1, 49, 50, 51]
    input_sell_ins = [-1, 0, 1, 10, 11, 12]

    verify_all_combinations(
        do_stuff,
        [
            input_names,
            input_sell_ins,
            input_qualities,
        ],
    )


def do_stuff(name, sell_in, quality):
    item = Item(name, sell_in, quality)
    GildedRose().update_quality([item])
    return str(item)
