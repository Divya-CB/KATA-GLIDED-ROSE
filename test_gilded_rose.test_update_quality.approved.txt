args: ('Normal', -1, -1) => 'Normal, -2, -1'
args: ('Normal', -1, 0) => 'Normal, -2, 0'
args: ('Normal', -1, 1) => 'Normal, -2, 0'
args: ('Normal', -1, 49) => 'Normal, -2, 47'
args: ('Normal', -1, 50) => 'Normal, -2, 48'
args: ('Normal', -1, 51) => 'Normal, -2, 49'
args: ('Normal', 0, -1) => 'Normal, -1, -1'
args: ('Normal', 0, 0) => 'Normal, -1, 0'
args: ('Normal', 0, 1) => 'Normal, -1, 0'
args: ('Normal', 0, 49) => 'Normal, -1, 47'
args: ('Normal', 0, 50) => 'Normal, -1, 48'
args: ('Normal', 0, 51) => 'Normal, -1, 49'
args: ('Normal', 1, -1) => 'Normal, 0, -1'
args: ('Normal', 1, 0) => 'Normal, 0, 0'
args: ('Normal', 1, 1) => 'Normal, 0, 0'
args: ('Normal', 1, 49) => 'Normal, 0, 48'
args: ('Normal', 1, 50) => 'Normal, 0, 49'
args: ('Normal', 1, 51) => 'Normal, 0, 50'
args: ('Normal', 10, -1) => 'Normal, 9, -1'
args: ('Normal', 10, 0) => 'Normal, 9, 0'
args: ('Normal', 10, 1) => 'Normal, 9, 0'
args: ('Normal', 10, 49) => 'Normal, 9, 48'
args: ('Normal', 10, 50) => 'Normal, 9, 49'
args: ('Normal', 10, 51) => 'Normal, 9, 50'
args: ('Normal', 11, -1) => 'Normal, 10, -1'
args: ('Normal', 11, 0) => 'Normal, 10, 0'
args: ('Normal', 11, 1) => 'Normal, 10, 0'
args: ('Normal', 11, 49) => 'Normal, 10, 48'
args: ('Normal', 11, 50) => 'Normal, 10, 49'
args: ('Normal', 11, 51) => 'Normal, 10, 50'
args: ('Normal', 12, -1) => 'Normal, 11, -1'
args: ('Normal', 12, 0) => 'Normal, 11, 0'
args: ('Normal', 12, 1) => 'Normal, 11, 0'
args: ('Normal', 12, 49) => 'Normal, 11, 48'
args: ('Normal', 12, 50) => 'Normal, 11, 49'
args: ('Normal', 12, 51) => 'Normal, 11, 50'
args: ('Conjured', -1, -1) => 'Conjured, -2, -1'
args: ('Conjured', -1, 0) => 'Conjured, -2, 0'
args: ('Conjured', -1, 1) => 'Conjured, -2, 0'
args: ('Conjured', -1, 49) => 'Conjured, -2, 47'
args: ('Conjured', -1, 50) => 'Conjured, -2, 48'
args: ('Conjured', -1, 51) => 'Conjured, -2, 49'
args: ('Conjured', 0, -1) => 'Conjured, -1, -1'
args: ('Conjured', 0, 0) => 'Conjured, -1, 0'
args: ('Conjured', 0, 1) => 'Conjured, -1, 0'
args: ('Conjured', 0, 49) => 'Conjured, -1, 47'
args: ('Conjured', 0, 50) => 'Conjured, -1, 48'
args: ('Conjured', 0, 51) => 'Conjured, -1, 49'
args: ('Conjured', 1, -1) => 'Conjured, 0, -1'
args: ('Conjured', 1, 0) => 'Conjured, 0, 0'
args: ('Conjured', 1, 1) => 'Conjured, 0, 0'
args: ('Conjured', 1, 49) => 'Conjured, 0, 48'
args: ('Conjured', 1, 50) => 'Conjured, 0, 49'
args: ('Conjured', 1, 51) => 'Conjured, 0, 50'
args: ('Conjured', 10, -1) => 'Conjured, 9, -1'
args: ('Conjured', 10, 0) => 'Conjured, 9, 0'
args: ('Conjured', 10, 1) => 'Conjured, 9, 0'
args: ('Conjured', 10, 49) => 'Conjured, 9, 48'
args: ('Conjured', 10, 50) => 'Conjured, 9, 49'
args: ('Conjured', 10, 51) => 'Conjured, 9, 50'
args: ('Conjured', 11, -1) => 'Conjured, 10, -1'
args: ('Conjured', 11, 0) => 'Conjured, 10, 0'
args: ('Conjured', 11, 1) => 'Conjured, 10, 0'
args: ('Conjured', 11, 49) => 'Conjured, 10, 48'
args: ('Conjured', 11, 50) => 'Conjured, 10, 49'
args: ('Conjured', 11, 51) => 'Conjured, 10, 50'
args: ('Conjured', 12, -1) => 'Conjured, 11, -1'
args: ('Conjured', 12, 0) => 'Conjured, 11, 0'
args: ('Conjured', 12, 1) => 'Conjured, 11, 0'
args: ('Conjured', 12, 49) => 'Conjured, 11, 48'
args: ('Conjured', 12, 50) => 'Conjured, 11, 49'
args: ('Conjured', 12, 51) => 'Conjured, 11, 50'
args: ('Aged Brie', -1, -1) => 'Aged Brie, -2, 1'
args: ('Aged Brie', -1, 0) => 'Aged Brie, -2, 2'
args: ('Aged Brie', -1, 1) => 'Aged Brie, -2, 3'
args: ('Aged Brie', -1, 49) => 'Aged Brie, -2, 50'
args: ('Aged Brie', -1, 50) => 'Aged Brie, -2, 50'
args: ('Aged Brie', -1, 51) => 'Aged Brie, -2, 51'
args: ('Aged Brie', 0, -1) => 'Aged Brie, -1, 1'
args: ('Aged Brie', 0, 0) => 'Aged Brie, -1, 2'
args: ('Aged Brie', 0, 1) => 'Aged Brie, -1, 3'
args: ('Aged Brie', 0, 49) => 'Aged Brie, -1, 50'
args: ('Aged Brie', 0, 50) => 'Aged Brie, -1, 50'
args: ('Aged Brie', 0, 51) => 'Aged Brie, -1, 51'
args: ('Aged Brie', 1, -1) => 'Aged Brie, 0, 0'
args: ('Aged Brie', 1, 0) => 'Aged Brie, 0, 1'
args: ('Aged Brie', 1, 1) => 'Aged Brie, 0, 2'
args: ('Aged Brie', 1, 49) => 'Aged Brie, 0, 50'
args: ('Aged Brie', 1, 50) => 'Aged Brie, 0, 50'
args: ('Aged Brie', 1, 51) => 'Aged Brie, 0, 51'
args: ('Aged Brie', 10, -1) => 'Aged Brie, 9, 0'
args: ('Aged Brie', 10, 0) => 'Aged Brie, 9, 1'
args: ('Aged Brie', 10, 1) => 'Aged Brie, 9, 2'
args: ('Aged Brie', 10, 49) => 'Aged Brie, 9, 50'
args: ('Aged Brie', 10, 50) => 'Aged Brie, 9, 50'
args: ('Aged Brie', 10, 51) => 'Aged Brie, 9, 51'
args: ('Aged Brie', 11, -1) => 'Aged Brie, 10, 0'
args: ('Aged Brie', 11, 0) => 'Aged Brie, 10, 1'
args: ('Aged Brie', 11, 1) => 'Aged Brie, 10, 2'
args: ('Aged Brie', 11, 49) => 'Aged Brie, 10, 50'
args: ('Aged Brie', 11, 50) => 'Aged Brie, 10, 50'
args: ('Aged Brie', 11, 51) => 'Aged Brie, 10, 51'
args: ('Aged Brie', 12, -1) => 'Aged Brie, 11, 0'
args: ('Aged Brie', 12, 0) => 'Aged Brie, 11, 1'
args: ('Aged Brie', 12, 1) => 'Aged Brie, 11, 2'
args: ('Aged Brie', 12, 49) => 'Aged Brie, 11, 50'
args: ('Aged Brie', 12, 50) => 'Aged Brie, 11, 50'
args: ('Aged Brie', 12, 51) => 'Aged Brie, 11, 51'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, -1) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, 0) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, 1) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, 49) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, 50) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', -1, 51) => 'Backstage passes to a TAFKAL80ETC concert, -2, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, -1) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, 0) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, 1) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, 49) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, 50) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 0, 51) => 'Backstage passes to a TAFKAL80ETC concert, -1, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, -1) => 'Backstage passes to a TAFKAL80ETC concert, 0, 2'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, 0) => 'Backstage passes to a TAFKAL80ETC concert, 0, 3'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, 1) => 'Backstage passes to a TAFKAL80ETC concert, 0, 4'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, 49) => 'Backstage passes to a TAFKAL80ETC concert, 0, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, 50) => 'Backstage passes to a TAFKAL80ETC concert, 0, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 1, 51) => 'Backstage passes to a TAFKAL80ETC concert, 0, 51'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, -1) => 'Backstage passes to a TAFKAL80ETC concert, 9, 1'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, 0) => 'Backstage passes to a TAFKAL80ETC concert, 9, 2'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, 1) => 'Backstage passes to a TAFKAL80ETC concert, 9, 3'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, 49) => 'Backstage passes to a TAFKAL80ETC concert, 9, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, 50) => 'Backstage passes to a TAFKAL80ETC concert, 9, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 10, 51) => 'Backstage passes to a TAFKAL80ETC concert, 9, 51'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, -1) => 'Backstage passes to a TAFKAL80ETC concert, 10, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, 0) => 'Backstage passes to a TAFKAL80ETC concert, 10, 1'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, 1) => 'Backstage passes to a TAFKAL80ETC concert, 10, 2'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, 49) => 'Backstage passes to a TAFKAL80ETC concert, 10, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, 50) => 'Backstage passes to a TAFKAL80ETC concert, 10, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 11, 51) => 'Backstage passes to a TAFKAL80ETC concert, 10, 51'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, -1) => 'Backstage passes to a TAFKAL80ETC concert, 11, 0'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, 0) => 'Backstage passes to a TAFKAL80ETC concert, 11, 1'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, 1) => 'Backstage passes to a TAFKAL80ETC concert, 11, 2'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, 49) => 'Backstage passes to a TAFKAL80ETC concert, 11, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, 50) => 'Backstage passes to a TAFKAL80ETC concert, 11, 50'
args: ('Backstage passes to a TAFKAL80ETC concert', 12, 51) => 'Backstage passes to a TAFKAL80ETC concert, 11, 51'
args: ('Sulfuras, Hand of Ragnaros', -1, -1) => 'Sulfuras, Hand of Ragnaros, -1, -1'
args: ('Sulfuras, Hand of Ragnaros', -1, 0) => 'Sulfuras, Hand of Ragnaros, -1, 0'
args: ('Sulfuras, Hand of Ragnaros', -1, 1) => 'Sulfuras, Hand of Ragnaros, -1, 1'
args: ('Sulfuras, Hand of Ragnaros', -1, 49) => 'Sulfuras, Hand of Ragnaros, -1, 49'
args: ('Sulfuras, Hand of Ragnaros', -1, 50) => 'Sulfuras, Hand of Ragnaros, -1, 50'
args: ('Sulfuras, Hand of Ragnaros', -1, 51) => 'Sulfuras, Hand of Ragnaros, -1, 51'
args: ('Sulfuras, Hand of Ragnaros', 0, -1) => 'Sulfuras, Hand of Ragnaros, 0, -1'
args: ('Sulfuras, Hand of Ragnaros', 0, 0) => 'Sulfuras, Hand of Ragnaros, 0, 0'
args: ('Sulfuras, Hand of Ragnaros', 0, 1) => 'Sulfuras, Hand of Ragnaros, 0, 1'
args: ('Sulfuras, Hand of Ragnaros', 0, 49) => 'Sulfuras, Hand of Ragnaros, 0, 49'
args: ('Sulfuras, Hand of Ragnaros', 0, 50) => 'Sulfuras, Hand of Ragnaros, 0, 50'
args: ('Sulfuras, Hand of Ragnaros', 0, 51) => 'Sulfuras, Hand of Ragnaros, 0, 51'
args: ('Sulfuras, Hand of Ragnaros', 1, -1) => 'Sulfuras, Hand of Ragnaros, 1, -1'
args: ('Sulfuras, Hand of Ragnaros', 1, 0) => 'Sulfuras, Hand of Ragnaros, 1, 0'
args: ('Sulfuras, Hand of Ragnaros', 1, 1) => 'Sulfuras, Hand of Ragnaros, 1, 1'
args: ('Sulfuras, Hand of Ragnaros', 1, 49) => 'Sulfuras, Hand of Ragnaros, 1, 49'
args: ('Sulfuras, Hand of Ragnaros', 1, 50) => 'Sulfuras, Hand of Ragnaros, 1, 50'
args: ('Sulfuras, Hand of Ragnaros', 1, 51) => 'Sulfuras, Hand of Ragnaros, 1, 51'
args: ('Sulfuras, Hand of Ragnaros', 10, -1) => 'Sulfuras, Hand of Ragnaros, 10, -1'
args: ('Sulfuras, Hand of Ragnaros', 10, 0) => 'Sulfuras, Hand of Ragnaros, 10, 0'
args: ('Sulfuras, Hand of Ragnaros', 10, 1) => 'Sulfuras, Hand of Ragnaros, 10, 1'
args: ('Sulfuras, Hand of Ragnaros', 10, 49) => 'Sulfuras, Hand of Ragnaros, 10, 49'
args: ('Sulfuras, Hand of Ragnaros', 10, 50) => 'Sulfuras, Hand of Ragnaros, 10, 50'
args: ('Sulfuras, Hand of Ragnaros', 10, 51) => 'Sulfuras, Hand of Ragnaros, 10, 51'
args: ('Sulfuras, Hand of Ragnaros', 11, -1) => 'Sulfuras, Hand of Ragnaros, 11, -1'
args: ('Sulfuras, Hand of Ragnaros', 11, 0) => 'Sulfuras, Hand of Ragnaros, 11, 0'
args: ('Sulfuras, Hand of Ragnaros', 11, 1) => 'Sulfuras, Hand of Ragnaros, 11, 1'
args: ('Sulfuras, Hand of Ragnaros', 11, 49) => 'Sulfuras, Hand of Ragnaros, 11, 49'
args: ('Sulfuras, Hand of Ragnaros', 11, 50) => 'Sulfuras, Hand of Ragnaros, 11, 50'
args: ('Sulfuras, Hand of Ragnaros', 11, 51) => 'Sulfuras, Hand of Ragnaros, 11, 51'
args: ('Sulfuras, Hand of Ragnaros', 12, -1) => 'Sulfuras, Hand of Ragnaros, 12, -1'
args: ('Sulfuras, Hand of Ragnaros', 12, 0) => 'Sulfuras, Hand of Ragnaros, 12, 0'
args: ('Sulfuras, Hand of Ragnaros', 12, 1) => 'Sulfuras, Hand of Ragnaros, 12, 1'
args: ('Sulfuras, Hand of Ragnaros', 12, 49) => 'Sulfuras, Hand of Ragnaros, 12, 49'
args: ('Sulfuras, Hand of Ragnaros', 12, 50) => 'Sulfuras, Hand of Ragnaros, 12, 50'
args: ('Sulfuras, Hand of Ragnaros', 12, 51) => 'Sulfuras, Hand of Ragnaros, 12, 51'
